src/app/cluster/list/component.ts

```diff
@@ -27,11 +27,15 @@ export class ClustersComponent implements OnInit, OnDestroy {
   private _unsubscribe: Subject<void> = new Subject<void>();
   areExternalClustersEnabled = false;
 
-  constructor(private readonly _settingsService: SettingsService, private readonly _clusterService: ClusterService) {}
+  constructor(private readonly _settingsService: SettingsService, private route: ActivatedRoute) {}
 
-  clusterIndex = this._clusterService.clusterIndex;
+  clusterIndex = '';
 
   ngOnInit(): void {
+    this.route.fragment.subscribe((fragment: string) => {
```

Comment:
It is a leak. Besides that, I'm not sure if we need to subscribe for that. If there is a option to get current value once and use it, I would do it.